src/TaskCreationSection/CategoryCreationSection/IconPicker/IconPicker.jsx

```diff
@@ -1,61 +1,65 @@
-/* eslint-disable no-use-before-define */
-/* eslint-disable import/prefer-default-export */
+/* eslint-disable react/no-unstable-nested-components */
 /* eslint-disable react/jsx-no-constructed-context-values */
 /* eslint-disable react/prop-types */
-import React, { useMemo, useState, useRef } from "react";
+import React, { useMemo } from "react";
 import { IconContext } from "react-icons";
 import * as icons from "react-icons/io";
 import { colorStyleInputHandler } from "../../../colorStyleClassHandler";
 import DynamicIcon from "./DynamicIcon";
 
-const toArray = (z) => Object.values(z);
-const getLimitIcons = (z, limit = 25) => toArray(z).slice(0, limit);
-const findIcons = (z, searchIcon) =>
-  icons.filter((icon) =>
-    icon.name.toLowerCase().includes(searchIcon.toLowerCase())
-  );
-const debounceTime = 300;
-
-export function IconPicker({
+function IconPicker({
   newCategoryIcon,
   setNewCategoryIcon,
   colorStyle,
+  searchIcon,
+  setSearchIcon,
 }) {
-  const [searchIcon, setSearchIcon] = useState("");
-  const [iconsToDisplay, setIconsToDisplay] = useState(toArray(icons));
-  const loadingIcons = useRef(false);
   const onInput = (e) => {
     setSearchIcon(e.target.value);
-    handleLoadingIcons(e.target.value);
-  };
-
-  const handleLoadingIcons = (value) => {
-    if (!value) {
-      setIconsToDisplay(toArray(icons));
-    } else if (!loadingIcons.current) {
-      loadingIcons.current = true;
-      setTimeout(() => {
-        setIconsToDisplay(findIcons(getLimitIcons(icons), searchIcon));
-        loadingIcons.current = false;
-      }, debounceTime);
-    }
   };
+  const memoizedResult = useMemo(() => {
+    return Object.entries(icons)
+      .filter(([name]) => name.toLowerCase().includes(searchIcon.toLowerCase()))
+      .map(([name, Icon]) => ({ name, Icon }));
+  }, [searchIcon]);
 
   const children = useMemo(
```

Comment:
we shouldn't assign component internally to variable in order to just render it later in the same component's return function because it decreases readability.

Remember memoization is not for free, the cost is additional complexity which decreases readability and because of that we increase possibility to introduce bug in the code 

Also general not written rule is that we use `children` name for the component that we want dynamically render inside another one, we also shouldn't use that when we don't have to, because of readability

e.g.
```
const CompA = () => {
  // return UI a
}
const CompB = () => {
  // return UI b
}

const ParentComp = ({ children }) => {
   return (
   <div>{children}</div>
  )

const Main = ({ name }) => {
  return (
  <div>
   <ParentComp>
      <CompA>
   </ParentComp>
   <ParentComp>
      <CompB>
    </ParentComp>
  </div>
  )
}
```